src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/EmptyLineWrappingInBlockCheck.java

```diff
@@ -0,0 +1,440 @@
+     */
+    public static final String MSG_SHOULD_BE_NO_EMPTY_LINE_AT_TOP = "block.no.empty.line.top";
+
+    /**
+     * A key is pointing to the warning message block.no.empty.line.bottom in "messages.properties"
+     * file.
+     */
+    public static final String MSG_SHOULD_BE_NO_EMPTY_LINE_AT_BOTTOM = "block.no.empty.line.bottom";
+
+    /** The policy to enforce at the beginning of the block. */
+    private BlockWrapOption topSeparator = BlockWrapOption.EMPTY_LINE_ALLOWED;
+
+    /** The policy to enforce at the end of the block. */
+    private BlockWrapOption bottomSeparator = BlockWrapOption.EMPTY_LINE_ALLOWED;
+
+    /**
+     * Sets the policy for the beginning of the block.
+     * @param optionStr string to decode option from
+     * @throws IllegalArgumentException if unable to decode
+     */
+    public void setTopSeparator(String optionStr) {
+        topSeparator = parseWrapOption(optionStr);
+    }
+
+    /**
+     * Sets the policy for the end of the block.
+     * @param optionStr string to decode option from
+     * @throws IllegalArgumentException if unable to decode
+     */
+    public void setBottomSeparator(String optionStr) {
+        bottomSeparator = parseWrapOption(optionStr);
+    }
+
+    @Override
+    public boolean isCommentNodesRequired() {
+        return true;
+    }
+
+    @Override
+    public int[] getDefaultTokens() {
+        return new int[] {
+            TokenTypes.ANNOTATION_DEF,
+            TokenTypes.CLASS_DEF,
+            TokenTypes.ENUM_DEF,
+            TokenTypes.INTERFACE_DEF,
+        };
+    }
+
+    @Override
+    public int[] getAcceptableTokens() {
+        return new int[] {
+            TokenTypes.ANNOTATION_ARRAY_INIT,
+            TokenTypes.ANNOTATION_DEF,
+            TokenTypes.ARRAY_INIT,
+            TokenTypes.CASE_GROUP,
+            TokenTypes.CLASS_DEF,
+            TokenTypes.CTOR_DEF,
+            TokenTypes.ENUM_CONSTANT_DEF,
+            TokenTypes.ENUM_DEF,
+            TokenTypes.INSTANCE_INIT,
+            TokenTypes.INTERFACE_DEF,
+            TokenTypes.LAMBDA,
+            TokenTypes.LITERAL_CASE,
+            TokenTypes.LITERAL_CATCH,
+            TokenTypes.LITERAL_DEFAULT,
+            TokenTypes.LITERAL_DO,
+            TokenTypes.LITERAL_ELSE,
+            TokenTypes.LITERAL_FINALLY,
+            TokenTypes.LITERAL_FOR,
+            TokenTypes.LITERAL_IF,
+            TokenTypes.LITERAL_NEW,
+            TokenTypes.LITERAL_SWITCH,
+            TokenTypes.LITERAL_SYNCHRONIZED,
+            TokenTypes.LITERAL_TRY,
+            TokenTypes.LITERAL_WHILE,
+            TokenTypes.METHOD_DEF,
+            TokenTypes.OBJBLOCK,
+            TokenTypes.SLIST,
+            TokenTypes.STATIC_INIT,
+        };
+    }
+
+    @Override
+    public int[] getRequiredTokens() {
+        return CommonUtils.EMPTY_INT_ARRAY;
+    }
+
+    @Override
+    public void visitToken(DetailAST ast) {
+
+        DetailAST leftBrace = null;
+        switch (ast.getType()) {
```

Comment:
please move all this huge Switch to method and return `leftBrace`. Nobody is interested on how you get this AST node - hide it in method.

Reply:
done.